images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vmbda_watcher.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watcher
+
+import (
+	"log/slog"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+)
+
+type VirtualMachineBlockDeviceAttachmentWatcher struct {
+	client client.Client
+	logger *slog.Logger
+}
+
+func NewVirtualMachineBlockDeviceAttachmentWatcher(logger *slog.Logger, client client.Client) *VirtualMachineBlockDeviceAttachmentWatcher {
+	return &VirtualMachineBlockDeviceAttachmentWatcher{
+		client: client,
+		logger: logger.With("watcher", "vmbda"),
+	}
+}
+
+func (w VirtualMachineBlockDeviceAttachmentWatcher) Run(mgr manager.Manager, ctr controller.Controller) error {
```

Comment:
Run is an ambiguous name: it overlaps with Run from SharedInformer. Why not Watch?

Reply:
mb Register?